cowbird/handlers/impl/filesystem.py

```diff
@@ -1,79 +1,449 @@
 import os
+import re
 import shutil
-from typing import Any
+from pathlib import Path
+from typing import Any, List, Tuple, cast
 
+from magpie.permissions import Access
+from magpie.permissions import Permission as MagpiePermission
+from magpie.services import ServiceAPI
+
+from cowbird.handlers import HandlerFactory
 from cowbird.handlers.handler import HANDLER_WORKSPACE_DIR_PARAM, Handler
+from cowbird.monitoring.fsmonitor import FSMonitor
+from cowbird.monitoring.monitoring import Monitoring
 from cowbird.permissions_synchronizer import Permission
-from cowbird.typedefs import SettingsType
-from cowbird.utils import get_logger
+from cowbird.typedefs import JSON, SettingsType
+from cowbird.utils import apply_new_path_permissions, get_logger
 
 LOGGER = get_logger(__name__)
 
 NOTEBOOKS_DIR_NAME = "notebooks"
+SECURE_DATA_PROXY_NAME = "secure-data-proxy"
+USER_WPS_OUTPUTS_USER_DIR_NAME = "wpsoutputs"
```

Comment:
Technically, `SECURE_DATA_PROXY_NAME` should not be needed if the component is not enabled in the stack, but I guess this is not something that matters a lot because the specific name is only relevant and used when exactly the same service is found on Magpie. 

I think they could both be optional in the config, but they must resolve to a "required"/defined `str` using those defaults for the code to work.

Reply:
I will put it as optional, with a default value if None.